packages/nextjs/src/config/wrappers/wrapperUtils.ts

```diff
@@ -1,58 +1,45 @@
 
 /**
- * Create a span to track the wrapped function and update transaction name with parameterized route.
+ * Call a data fetcher and trace it. Only traces the function if there is an active transaction on the scope.
  *
- * @template T Types for `getInitialProps`, `getStaticProps`, and `getServerSideProps`
- * @param origFunction The user's exported `getInitialProps`, `getStaticProps`, or `getServerSideProps` function
- * @param context The context object passed by nextjs to the function
- * @param route The route currently being served
- * @returns The result of calling the user's function
+ * We only do the following until we move transaction creation into this function: When called, the wrapped function
+ * will also update the name of the active transaction with a parameterized route provided via the `options` argument.
  */
-export async function wrapperCore<T extends DataFetchingFunction>(
-  origFunction: T['fn'],
-  context: T['context'],
-  route: string,
-): Promise<T['result']> {
-  const transaction = getActiveTransaction();
-
-  if (transaction) {
-    // Pull off any leading underscores we've added in the process of wrapping the function
-    const wrappedFunctionName = origFunction.name.replace(/^_*/, '');
-
-    // TODO: Make sure that the given route matches the name of the active transaction (to prevent background data
-    // fetching from switching the name to a completely other route)
-    transaction.name = route;
-    transaction.metadata.source = 'route';
+export async function callDataFetcherTraced<F extends (...args: any[]) => Promise<any> | any>(
+  origFunction: F,
+  origFunctionArgs: Parameters<F>,
+  options: {
+    parameterizedRoute: string;
+    dataFetchingMethodName: string;
+  },
+): Promise<ReturnType<F>> {
+  const { parameterizedRoute, dataFetchingMethodName } = options;
 
-    // Capture the route, since pre-loading, revalidation, etc might mean that this span may happen during another
-    // route's transaction
-    const span = transaction.startChild({ op: 'nextjs.data', description: `${wrappedFunctionName} (${route})` });
-
-    const props = await callOriginal(origFunction, context, span);
-
-    span.finish();
+  const transaction = getActiveTransaction();
 
-    return props;
+  if (!transaction) {
+    return origFunction(...origFunctionArgs);
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  return callOriginal(origFunction, context);
-}
+  // TODO: Make sure that the given route matches the name of the active transaction (to prevent background data
+  // fetching from switching the name to a completely other route) -- We'll probably switch to creating a transaction
+  // right here so making that check will probabably not even be necessary.
+  // Logic will be: If there is no active transaction, start one with correct name and source. If there is an active
+  // transaction, create a child span with correct name and source.
```

Comment:
Just for me to get this right: If we already have a transaction, this means that we started it beforehand in a previous data fetcher function (iirc you said that getInitialProps can e.g. be executed before getServerSideProps), right?

Reply:
Right now it's started in `instrumentServer.ts` - but actually, that's not working so there never is a transaction. Soooo.... this PR currently doesn't do anything lol.

Reply:
Yeah, something is weird, because there IS in fact a transaction in our integration tests, otherwise they wouldn't pass. And there's a transaction if you run the server inside of the debugger. It's very mysterious.